File: src/main/java/com/prova/swapi/model/Nave.java
package com.prova.swapi.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
/**
 * Representa uma pessoa.
 */
@Data
public class Nave {
    private String name;
    private String model;
    private String passengers;
    private String manufacturer;
    @JsonProperty("cost_in_credits")
    private String costInCredits;
    private String length;
    @JsonProperty("max_atmosphering_speed")
    private String maxAtmospheringSpeed;
    private String crew;
    @JsonProperty("cargo_capacity")
    private String cargoCapacity;
    @JsonProperty("MGLT")
    private String mglt;
    @JsonProperty("starship_class")
    private String starshipClass;
    private Double autonomia;

}

File: swfrontend/src/app/components/naves/models/nave.ts
export class Nave {

    nome: string;
    modelo: number;
    qtdPassageiros: string;
    manufaturado: string;
    valor: string;
    tamanho: string;
    velocidadeAtsmofericaMax: string;
    tamEquipe: string;
    capCarga: string;
    mglt: string;
    classe: string;
    tempo: number;

    constructor(item) {
        this.nome = item.name;
        this.modelo = item.model;
        this.qtdPassageiros = item.passengers;
        this.manufaturado = item.manufacturer;
        this.valor = item.cost_in_credits;
        this.tamanho = item.length;
        this.velocidadeAtsmofericaMax = item.max_atmosphering_speed;
        this.tamEquipe = item.crew;
        this.capCarga = item.cargo_capacity;
        this.mglt = item.MGLT;
        this.classe = item.starship_class;
    }
}
